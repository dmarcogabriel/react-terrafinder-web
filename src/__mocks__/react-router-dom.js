const originalModule = jest.requireActual('react-router-dom');

module.exports = {
  ...originalModule,
  useHistory: () => ({
    history: { push: jest.fn() },
  }),
};
