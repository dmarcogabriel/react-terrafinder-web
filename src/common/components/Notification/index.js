import React from 'react';
import { useNotification } from 'hooks/useNotification';
import { Alert, Snackbar } from '@mui/material';

export default function Notification() {
  const { notification, show } = useNotification();

  return (
    <Snackbar open={show} autoHideDuration={5000} data-testid="notification">
      <Alert
        variant="filled"
        elevation={6}
        severity={notification.type}
        sx={{ width: '100%' }}
        data-testid="notificationMessage"
      >
        {notification.message}
      </Alert>
    </Snackbar>
  );
}
