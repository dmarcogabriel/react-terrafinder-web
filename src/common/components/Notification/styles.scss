@import '~styles';

.Notification__container {
  position: fixed;
  top: 2rem;
  z-index: 5000;
  right: 2rem;
  left: 2rem;
  padding: 1rem;
  border-radius: 0.25rem;
  opacity: 0.8;
  transition: opacity 5s;

  > p {
    opacity: 1;
    text-align: center;
  }
}

.success {
  background: $green-light;
  color: $green-dark;
}

.info {
  background: $blue;
  color: $blue-dark;
}

.warning {
  background: $primary-light;
  color: $orange;
}

.error {
  background: $red;
  color: $red-dark;
}
