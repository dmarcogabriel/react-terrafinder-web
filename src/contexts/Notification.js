import React, { createContext, useState, useEffect } from 'react';

export const Context = createContext();

export const NotificationProvider = ({ children }) => {
  const [show, setShow] = useState(false);
  const [notification, setNotification] = useState({
    message: '',
    type: 'info',
  });

  const showNotification = (message, type = 'info') => {
    setNotification({ message, type });
    setShow(true);
  };

  useEffect(() => {
    if (show) {
      setTimeout(() => {
        setShow(false);
      }, 5000);
    }
  }, [show]);

  return (
    <Context.Provider
      value={{
        show,
        notification,
        showNotification,
      }}
    >
      {children}
    </Context.Provider>
  );
};
