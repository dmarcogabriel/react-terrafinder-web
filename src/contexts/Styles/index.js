import React from 'react';
import { ThemeProvider as Provider } from 'styled-components';
import defaultTheme from './themes/default';

export default function ThemeProvider({ children }) {
  return <Provider theme={defaultTheme}>{children}</Provider>;
}
