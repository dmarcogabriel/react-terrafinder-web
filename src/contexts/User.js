import React, { createContext, useState, useEffect } from 'react';
import { USER_KEY } from 'constants/userKey';
import api from 'services/api';

export const Context = createContext();

export const UserProvider = ({ children, storedUser }) => {
  const [currentUser, setCurrentUser] = useState(null);
  const [loadingUser, setLoadingUser] = useState();

  const login = ({ token, email, firstName, _id, ...user }) => {
    const data = { _id, firstName, email, token };

    window.localStorage.setItem(USER_KEY, JSON.stringify(data));
    setCurrentUser({ ...user, ...data });
  };

  const userIsLogged = () => !!currentUser;

  const logout = () => {
    window.localStorage.removeItem(USER_KEY);
    setCurrentUser(null);
    window.location.href = '/home';
  };

  const loadUserData = async () => {
    setLoadingUser(true);

    try {
      const { data: res } = await api.get(`users/${storedUser._id}`);
      console.log(res.data);
      setCurrentUser({ ...res.data.user, token: storedUser.token });
      setLoadingUser(false);
    } catch (err) {
      logout();
      setLoadingUser(false);
    }
  };

  const updateUser = (data) => {
    setCurrentUser((old) => ({ ...old, ...data }));
  };

  useEffect(() => {
    if (storedUser) loadUserData();
  }, []);

  return (
    <Context.Provider
      value={{ currentUser, login, logout, updateUser, userIsLogged }}
    >
      {!loadingUser && children}
    </Context.Provider>
  );
};
