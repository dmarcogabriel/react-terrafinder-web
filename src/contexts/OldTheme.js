import React, { useState } from 'react';
import { ThemeProvider as StyledComponentsProvider } from 'styled-components';
import { theme as defaultTheme } from 'styles/themes';

export const OldThemeProvider = ({ children }) => {
  const [theme] = useState(defaultTheme);

  return (
    <StyledComponentsProvider theme={theme}>
      {children}
    </StyledComponentsProvider>
  );
};
