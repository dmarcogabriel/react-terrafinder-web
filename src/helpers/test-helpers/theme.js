import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/themes';

export const withTheme = (component) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);

export const renderWithTheme = (component) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
