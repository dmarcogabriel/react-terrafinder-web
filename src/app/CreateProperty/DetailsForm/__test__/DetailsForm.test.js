import React from 'react';
import ReactDOM from 'react-dom';
import { render, fireEvent } from '@testing-library/react';
import { UserProvider } from 'contexts/User';
import { NotificationProvider } from 'contexts/Notification';
import { BrowserRouter as Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { renderWithTheme, withTheme } from 'helpers/test-helpers/theme';
import { DetailsForm } from '..';

const Comp = () => (
  <UserProvider>
    <NotificationProvider>
      <Router>
        <DetailsForm />
      </Router>
    </NotificationProvider>
  </UserProvider>
);

describe('app/CreateProperty/DetailsForm', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(withTheme(<Comp />), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderWithTheme(<Comp />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should fires go back event', () => {
    const history = createMemoryHistory();

    const { getByTestId } = render(withTheme(<Comp />));

    const backButton = getByTestId('back');
    fireEvent.click(backButton);

    expect(history.location.pathname).toEqual('/');
  });

  describe('activity', () => {
    it('should add activity', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const addActivityButton = getByTestId('add-activity-button');
      fireEvent.click(addActivityButton);

      const activity = getByTestId('activity-input-1');
      expect(activity).toBeInTheDocument();
    });

    it('should change activity value', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const actInput = getByTestId('activity-input-0');
      fireEvent.change(actInput, { target: { value: 'testing' } });

      expect(actInput).toHaveValue('testing');
    });

    it('should add 5 activities', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const addActivityButton = getByTestId('add-activity-button');

      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);

      const acts = getByTestId('activities');
      const lastActivity = getByTestId('activity-input-4');
      expect(lastActivity).toBeInTheDocument();
      expect(acts.childElementCount).toEqual(6);
    });

    it('should fail to add more than 5 activities', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const addActivityButton = getByTestId('add-activity-button');

      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);

      const acts = getByTestId('activities');
      expect(acts.childElementCount).toEqual(6);
    });

    it('should change the right input', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const addActivityButton = getByTestId('add-activity-button');

      fireEvent.click(addActivityButton);
      fireEvent.click(addActivityButton);

      const selectedInput = getByTestId('activity-input-1');
      fireEvent.change(selectedInput, { target: { value: 'testing' } });

      expect(selectedInput).toHaveValue('testing');
      expect(getByTestId('activity-input-0')).toHaveValue('');
      expect(getByTestId('activity-input-2')).toHaveValue('');
    });
  });

  describe('farming', () => {
    it('should add farming', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const addFarming = getByTestId('add-farming-button');
      fireEvent.click(addFarming);

      const farm = getByTestId('farming-input-1');
      expect(farm).toBeInTheDocument();
    });

    it('should add 5 farmings', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const addFarming = getByTestId('add-farming-button');
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);

      const farms = getByTestId('farms');
      const farm = getByTestId('farming-input-4');
      expect(farm).toBeInTheDocument();
      expect(farms.childElementCount).toEqual(6);
    });

    it('should fail fo add more than 5 farmings', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const addFarming = getByTestId('add-farming-button');
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);
      fireEvent.click(addFarming);

      const farms = getByTestId('farms');
      expect(farms.childElementCount).toEqual(6);
    });

    it('should change farming value', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const farmInput = getByTestId('farming-input-0');
      fireEvent.change(farmInput, { target: { value: 'testing' } });

      expect(farmInput).toHaveValue('testing');
    });

    it('should change the right input', () => {
      const { getByTestId } = render(withTheme(<Comp />));

      const addFarmingButton = getByTestId('add-farming-button');

      fireEvent.click(addFarmingButton);
      fireEvent.click(addFarmingButton);

      const selectedInput = getByTestId('farming-input-1');
      fireEvent.change(selectedInput, { target: { value: 'testing' } });

      expect(selectedInput).toHaveValue('testing');
      expect(getByTestId('farming-input-0')).toHaveValue('');
      expect(getByTestId('farming-input-2')).toHaveValue('');
    });
  });
});
