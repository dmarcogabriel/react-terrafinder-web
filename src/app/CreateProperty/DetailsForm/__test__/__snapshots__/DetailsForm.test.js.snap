// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/CreateProperty/DetailsForm matches snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiBox-root css-1663l76"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
            >
              Quanto quer pela propriedade?
            </label>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
              data-testid="MoodIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
              />
            </svg>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                >
                  R$
                </p>
              </div>
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                maxLength={14}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1a1ihhj"
                >
                  <span
                    className="notranslate"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiBox-root css-1663l76"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
            >
              Qual o tamanho da propriedade (em equitares)
            </label>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
              data-testid="MoodIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
              />
            </svg>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                maxLength={4}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1a1ihhj"
                >
                  <span
                    className="notranslate"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-1n15j52"
      >
        <div
          className="MuiBox-root css-0"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Cultivo (plantações)
          </p>
          <div
            className="MuiBox-root css-0"
            data-testid="farms"
          >
            <div
              className="MuiBox-root css-0"
            >
              <div
                className="MuiBox-root css-1663l76"
              >
                <label
                  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                >
                  Cultivo #1
                </label>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
                  data-testid="MoodIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
                  />
                </svg>
              </div>
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="farming-input-0"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-1a1ihhj"
                    >
                      <span
                        className="notranslate"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
                
              </div>
            </div>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSuccess MuiIconButton-sizeMedium css-1r6h6t4-MuiButtonBase-root-MuiIconButton-root"
              data-testid="add-farming-button"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
              variant="contained"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Atividades e destaques
          </p>
          <div
            className="MuiBox-root css-0"
            data-testid="activities"
          >
            <div
              className="MuiBox-root css-0"
            >
              <div
                className="MuiBox-root css-1663l76"
              >
                <label
                  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                >
                  Atividade #1
                </label>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
                  data-testid="MoodIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
                  />
                </svg>
              </div>
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="activity-input-0"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-1a1ihhj"
                    >
                      <span
                        className="notranslate"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
                
              </div>
            </div>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSuccess MuiIconButton-sizeMedium css-1r6h6t4-MuiButtonBase-root-MuiIconButton-root"
              data-testid="add-activity-button"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
              variant="contained"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-1795tmz"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1gxgiww-MuiButtonBase-root-MuiButton-root"
          data-testid="back"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Voltar
        </button>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-18x86fu-MuiButtonBase-root-MuiButton-root"
          data-testid="next"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Cadastrar
        </button>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div>
        <p>
          1
        </p>
        <div>
          <p>
            Escolha seu plano
          </p>
          <p>
            Explore as possibilidades dentro da plataforma
          </p>
        </div>
      </div>
      <div>
        <p>
          2
        </p>
        <div>
          <p>
            Informações básicas
          </p>
          <p>
            Nos fale um pouco mais sobre o seu imóvel
          </p>
        </div>
      </div>
      <div>
        <p>
          3
        </p>
        <div>
          <p>
            Informações Detalhadas
          </p>
          <p>
            Conte-nos mais detalhes sobre sua propriedade, para chamar mais a atenção
          </p>
        </div>
      </div>
      <div>
        <p>
          4
        </p>
        <div>
          <p>
            Conteúdos
          </p>
          <p>
            Você pode enviar imagens e vídeos da sua propriedade
          </p>
        </div>
      </div>
      <div>
        <p>
          5
        </p>
        <div>
          <p>
            Revisão
          </p>
          <p>
            Confirme o cadastro do seu anúncio
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
